src/coreclr/src/jit/valuenum.cpp

```diff
@@ -3614,6 +3614,13 @@ ValueNum ValueNumStore::VNApplySelectorsTypeCheck(ValueNum elem, var_types indTy
 
         if ((indType == TYP_REF) && (varTypeIsStruct(elemTyp)))
         {
+            // This whole block is over-optimistic, we don't have any information
```

Comment:
I checked that if we delete that unsafe condition we will get 8 bytes regression in framework assemblies (1 method) and 2 in SPMI (pri1 and BING).
That regression will be fixed with #33225, so I think I will just delete this block before the merge.